### ISSUE TITLE
[bug] Autowall Only visible  ignore glass

### ISSUE THREAD

**Issue by POWERzzzzz** [NullHooks/NullHooks#72]
### Bug description
Autowall only visible should ignore glas (maps like train, office etc.)

idk if this is possible at all but if it is then it'd be cool

**Comment by r4v10l1** [comment]
I agree, but I don't know how easy it would be to make. I will try to add it.

**Comment by Ceaco1337** [comment]
> I agree, but I don't know how easy it would be to make. I will try to add it.

Rebuild your ray trace within `can_see_player_pos` inside of  `entities.hpp`.
Look to see if your trace hits an entity and if that entity `is breakable`.

If you had a more reversed autowall, you'd understand it better. 
But currently you are just pattern scanning for functions like `trace_to_exit` instead.
https://github.com/NullHooks/NullHooks/blob/4461caf48238723033a9fc77b128a187e05bdff6/src/core/features/aim/autowall.cpp#L29

I would personally rebuild autowall. `(trace_to_exit, clip_trace_to_players, etc)`
But I know it's frustrating and also requires your interfaces and sdk to be more up-to-date.

If you are gonna pattern scan:
```cpp
"client.dll", "55 8B EC 51 56 8B F1 85 F6 74 68 83 BE"
```

Look into:
https://github.com/sstokic-tgm/Gladiatorcheatz-v2.1/blob/a075ea874b9d8077c6bec1c7079bd43e6cf16a6d/features/AimRage.cpp#L761

Complete Autowall:
https://github.com/rollraw/qo0-base/blob/master/base/features/autowall.cpp

Though, I'd focus on how these rays work:
https://github.com/rollraw/qo0-base/blob/cb140182ee647dc3fd9ebcf156f29b1d7457ad95/base/features/autowall.cpp#L154
